File: DongPoTea.Web/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  nameChinese?: string;
  description?: string;
  descriptionChinese?: string;
  sku: string;
  price: number;
  stockQuantity: number;
  weight?: number;
  inStock: boolean;
  category?: Category;
  primaryImage?: ProductImage;
  images: ProductImage[];
}

export interface ProductListItem {
  id: string;
  name: string;
  description?: string;
  price: number;
  inStock: boolean;
  category?: Category;
  primaryImage?: ProductImage;
}

export interface Category {
  id: string;
  name: string;
  nameChinese?: string;
  slug: string;
  productCount: number;
}

export interface ProductImage {
  id: string;
  url: string;
  altText?: string;
  isPrimary: boolean;
}

export interface ProductFilters {
  category?: string;
  search?: string;
  sortBy?: string;
  sortOrder?: 'asc' | 'desc';
  page?: number;
  pageSize?: number;
}

export interface PaginatedResponse<T> {
  data: T[];
  pagination: {
    currentPage: number;
    pageSize: number;
    totalPages: number;
    totalItems: number;
  };
}

File: DongPoTea.Core/Entities/Product.cs
namespace DongPoTea.Core.Entities;

public class Product : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? NameChinese { get; set; }
    public string? Description { get; set; }
    public string? DescriptionChinese { get; set; }
    public string SKU { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public Guid CategoryId { get; set; }
    public int StockQuantity { get; set; } = 0;
    public decimal? Weight { get; set; }
    public bool IsActive { get; set; } = true;
    
    public Category Category { get; set; } = null!;
    public ICollection<ProductImage> Images { get; set; } = new List<ProductImage>();
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public ICollection<CartItem> CartItems { get; set; } = new List<CartItem>();
}